snmp/datadog_checks/snmp/snmp.py

```diff
@@ -531,19 +546,12 @@ def submit_metric(self, name, snmp_value, forced_type, tags, options):
             metric_name = self.normalize(name, prefix='snmp')
 
         if forced_type is not None:
-            try:
-                metric = as_metric_with_forced_type(snmp_value, forced_type, options)
-            except Exception as e:
-                self.log.error('Unable to coerce %s to %s: %s', name, forced_type, e)
-                return
-            if metric is None:
-                raise ConfigurationError('Invalid forced-type {!r} for metric {!r}'.format(forced_type, name))
+            metric = as_metric_with_forced_type(snmp_value, forced_type, options)
         else:
             metric = as_metric_with_inferred_type(snmp_value)
 
         if metric is None:
-            self.log.warning('Unsupported metric type %s for %s', snmp_value.__class__.__name__, metric_name)
-            return
+            raise RuntimeError('Unsupported metric type {} for {}'.format(type(snmp_value), metric_name))
```

Comment:
Using `type` is safer, in case in case `snmp_value.__class__.__name__` cannot be fetched e.g. None type.